### PR TITLE
Add nametag background setting and object property

### PR DESCRIPTION
**Pull request by rubenwardy**
There's been some complaints that the new nametag background (#10152) breaks compatibility, and may be seen as controversial by modders 

This PR:

* Adds a setting to disable the default nametag background.
* Adds an object property for mods to set the nametag background on an per-entity/player basis.
* Adds an `optional` class to make the code a bit cleaner.

## To do

- [ ] Testing

## How to test

    /spawnentity testentities:nametag

### CODE REVIEW

**Comment by KaylebJay** [comment]
I don't like it at all personally, so thank you very much for this PR. :thumbsup:

File: src/object_properties.cpp [hunk 1]
```diff
@@ -120,6 +127,13 @@ void ObjectProperties::serialize(std::ostream &os) const
 	os << serializeString16(damage_texture_modifier);
 	writeU8(os, shaded);
 	writeU8(os, show_on_minimap);
+	if (nametag_bgcolor) {
+		writeU8(os, 1);
+		writeARGB8(os, nametag_bgcolor.value());
+	} else {
+		writeU8(os, 0);
+		writeARGB8(os, {});
+	}
```
Comment (sfan5) [inline src/object_properties.cpp:136]: it'd be possible to save one byte here by encoding `nullopt` as `0,1,1,1` and any alpha=0 color as `0,0,0,0`

alternatively: why write a color when it's known that there isn't one?

File: src/util/optional.h [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		return m_value;
+	}
+
+	const T &value_or(const T &def) const
+	{
+		if (m_has_value) {
+			return m_value;
+		} else {
+			return def;
+		}
```
Comment (sfan5) [inline src/util/optional.h:78]:
```suggestion
		return m_has_value ? m_value : def;
```

File: src/util/optional.h [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+/**
+ * An implementation of optional for C++11, which aims to be
+ * compatible with a subset of std::optional features.
+ *
+ * Unfortunately, Minetest doesn't use C++17 yet.
+ *
+ * @tparam T The type to be stored
+ */
+template <typename T>
+class optional
```
Comment (sfan5) [inline src/util/optional.h:38]: Somehow a lowercase class name looks weird but I don't know if `Optional<T>` would be better

Comment (rubenwardy) [reply src/util/optional.h:38]: Well, I think it should either be in `stdx` or start with a capital. I don't know which is better

Comment (rubenwardy) [reply src/util/optional.h]: using stdx is undefined behaviour, apparently

**Review by sfan5 — APPROVED** [review]
Looks good.

**Pull request merged**